ESQL: Tests for inlinestats with union types

astefan opened the pull request:
Addresses https://github.com/elastic/elasticsearch/issues/124731

astefan committed:
Tests for inlinestats with union types

astefan committed:
Enable CCQ

astefan committed:
Merge branch 'main' of https://github.com/elastic/elasticsearch into inlinestats_with_union_types

astefan committed:
Update

Commit:
[CI] Auto commit changes from spotless

astefan committed:
Disable CCQ

astefan committed:
Merge branch 'inlinestats_with_union_types' of https://github.com/astefan/elasticsearch into inlinestats_with_union_types

astefan committed:
Muted an inlinestats + fork IT test

astefan committed:
Merge branch 'main' of https://github.com/elastic/elasticsearch into inlinestats_with_union_types

astefan commented on x-pack/plugin/esql/qa/testFixtures/src/main/resources/union_types.csv-spec:
```diff
@@ -1398,62 +1398,6 @@ event_duration:long | _index:keyword       | ts:date                   | ts_str:
 ;
 
 
-inlineStatsUnionGroup-Ignore
```
These were moved to the `inlinestats` test file.

astefan committed:
Fix test after https://github.com/elastic/elasticsearch/pull/133651 merge

bpintea reviewed (approved):
Nice and broad set of tests, thanks.
I'd maybe add (at least) one where the union type is explicitly overwritten before the INLINESTATS. ~(There are some implicit EVALs before, extracted out of the STATS).~

astefan commented on muted-tests.yml:
```diff
@@ -531,6 +531,9 @@ tests:
   method: testInferClassificationModel
   issue: https://github.com/elastic/elasticsearch/issues/133448
+- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeForkIT
+  method: test {csv-spec:inlinestats.MultiIndexInlinestats OfMultiTypedField}
```
Because this one fails **only** in the presence of `fork` **and** as part of the GenerativeForkIT suite which automatically gathers csv-spec tests and adds `| FORK (WHERE true) (WHERE true)` to the end of the tests. This test runs fine in other IT/unit test setups.

astefan commented on x-pack/plugin/esql/qa/testFixtures/src/main/resources/inlinestats.csv-spec:
```diff
@@ -2480,3 +2480,1105 @@ from employees
+required_capability: inlinestats_v11
+
+FROM sample_data
+| EVAL client_ip = client_ip::ip
```
These tests are a "mirror" of what we already have for `stats` + union types.

astefan committed:
Address reviews:
- added comments to some of the csv-spec tests
- de-Ignored few csv-spec tests
- renamed (removed the white spaces) csv-spec tests
- more tests

astefan commented on x-pack/plugin/esql/qa/testFixtures/src/main/resources/inlinestats.csv-spec:
```diff
@@ -2480,3 +2480,1105 @@ from employees
+null                  |172.21.2.113   |2764889             |Connected to 10.1.0.2|2                   |14
+;
+
+multiIndexIpInlinestats PushableCount-Ignore
```
You are correct. Thank you for mentioning this, I forgot about them being ignored. Updated.

astefan committed:
Merge branch 'main' of https://github.com/elastic/elasticsearch into inlinestats_with_union_types

astefan committed:
Merge branch 'main' of https://github.com/elastic/elasticsearch into inlinestats_with_union_types

astefan committed:
make the test more deterministic

astefan committed:
Merge branch 'main' of https://github.com/elastic/elasticsearch into inlinestats_with_union_types

astefan committed:
more deterministic tests

astefan committed:
Merge branch 'main' of https://github.com/elastic/elasticsearch into inlinestats_with_union_types

astefan committed:
More tests adjustments

astefan committed:
Merge branch 'main' into inlinestats_with_union_types
